### PR TITLE
uri documentation update

### PR DESCRIPTION
**Pull request by burner**
* moved some examples to unittests
* made a only privately used enum private
* URIException made up to spec

### CODE REVIEW

File: std/uri.d [hunk 1]
```diff
@@ -38,20 +38,21 @@ private import std.traits : isSomeChar;
 import core.exception : OutOfMemoryError;
 import std.exception : assumeUnique;
 
+/** This Exception is thrown if the something goes wrong when encoding or
+decoding and uri.
```
Comment (quickfur) [inline std/uri.d]: Grammar: `... when encoding or decoding a URI`. (Before "u" the article "a" does not become "an" in many cases.)

Comment (burner) [inline std/uri.d]: one moment please

Comment (burner) [inline std/uri.d]: done

**Comment by quickfur** [comment]
LGTM, thanks!

**Comment by quickfur** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks for the review

**Pull request merged**